build/vsts-helpers.ps1

```diff
@@ -13,46 +12,27 @@
 		$Colour
 	)
 	
-	# Store the colour to be able to revert back to it.
-	$previousColour = [System.Console]::ForegroundColor
-	
-	# Set the appropriate foreground text colour.
-	switch ($Colour)
-	{
-		"Red" 		{ [System.Console]::ForegroundColor = [System.ConsoleColor]::Red }
-		"Green" 	{ [System.Console]::ForegroundColor = [System.ConsoleColor]::Green }
-		"Blue" 		{ [System.Console]::ForegroundColor = [System.ConsoleColor]::Blue }
-		"Cyan" 		{ [System.Console]::ForegroundColor = [System.ConsoleColor]::Cyan }
-		"Magenta" 	{ [System.Console]::ForegroundColor = [System.ConsoleColor]::Magenta }
-		"Yellow" 	{ [System.Console]::ForegroundColor = [System.ConsoleColor]::Yellow }
-		"Bold" 		{ [System.Console]::ForegroundColor = [System.ConsoleColor]::White }
-		"Black" 	{ [System.Console]::ForegroundColor = [System.ConsoleColor]::Black }
-	}
+	# Initialize the variable to output.
+	$HelpMessage = ''
 	
 	# Add some extra spacing to either side of the message.
 	$length = $Message.Length + 2
 	
-	# Write the top border.
-	Write-Host "+" -NoNewline
-	for ($i = 0; $i -lt $length; $i++)
-	{
-		Write-Host "-" -NoNewline
+	function WriteBorder($remark){
+		$HelpMessage += "+"
+		for ($i = 0; $i -lt $length; $i++)
+		{
+			$HelpMessage += "-"
+		}
+		$HelpMessage += "+$([Environment]::NewLine)" 
 	}
-	Write-Host "+"
+
+	. WriteBorder 'Top'
 	
-	# Write the text.
-	Write-Host "| " -NoNewline
-	Write-Host $Message -NoNewline
-	Write-Host " |"
+	$HelpMessage += "| $Message |$([Environment]::NewLine)" 
```

Comment:
Is there a reason why you used `$([Environment]::NewLine)` as opposed to `` `n ``? I'm curious if there is because i always use the latter purely because it's easier to read.

Reply:
> Is there a reason why you used `$([Environment]::NewLine)` as opposed to `` `n ``? I'm curious if there is because i always use the latter purely because it's easier to read.

the 'Newline' corresponds to different characters on different platforms, so using attributes instead of literals is automatically compatible.

> `\r\n` for non-Unix platforms, or `\n` for Unix platforms.

[Environment.NewLine Property (System) | Microsoft Docs](https://docs.microsoft.com/en-us/dotnet/api/system.environment.newline?view=net-5.0)

Reply:
Ah right, I wasn't aware of that. Cool, that makes sense.